src/plugins/discover/public/application/components/types.ts

```diff
@@ -100,6 +100,22 @@ export interface DiscoverProps {
      * Client of uiSettings
      */
     config: IUiSettingsClient;
+    /**
+     * returns field statistics based on the loaded data sample
+     */
+    getFieldCounts: () => Promise<Record<string, number>>;
```

Comment:
Why do we need these props here new?

Reply:
we need it because I had to inline the `getTopNavLinks` function , which also deangularizes stuff here a bit more. will be more than happy once we can also inline this function ... 

https://github.com/kertal/kibana/blob/6fa9c2af8fca6f16ad16d9c357a41a0cf62dcb86/src/plugins/discover/public/application/components/discover.tsx#L99-L99